src/components/FileUpload/FileUpload.js

```diff
@@ -1,12 +1,77 @@
-//reference: https://www.positronx.io/react-file-upload-tutorial-with-node-express-and-multer/
-import React, { Component } from 'react';
-import { FormGroup } from '../Utils/Utils';
-export default class FilesUploadComponent extends Component {
-  render() {
-    return (
-            <div className="form-group">
-              <input type="file" name="uploaded_file" onChange={this.props.onChange} />
-            </div>
-    )
-  }
-}
+import React, { useState } from "react";
+import Cropper from "react-cropper";
+import "cropperjs/dist/cropper.css";
+import "./FileUpload.css";
+
+export default function FileUpload(props) {
+  const { handler, imgHandler } = props;
+
+  const [image, setImage] = useState();
+  const [saved, setSaved ] = useState("");
+  const [showCropImage, setCropImage ] = useState("hide");
+  const [cropData, setCropData] = useState("#");
+  const [cropper, setCropper] = useState();
+  const onChange = (e) => {
+    e.preventDefault();
+    let files;
+    
+    if (e.dataTransfer) {
+      files = e.dataTransfer.files;
+    } else if (e.target) {
+      files = e.target.files;
+    }
+    const reader = new FileReader();
+    reader.onload = () => {
+      setImage(reader.result);
+    };
+    reader.readAsDataURL(files[0]);
+
+    setSaved("")
+    setCropImage("")
+    imgHandler(false)
+  };
+
+  const getCropData = () => {
+
+    if (typeof cropper !== "undefined") {
+      setCropData(cropper.getCroppedCanvas().toDataURL("image/jpeg"));
+    }
+
+    cropper.getCroppedCanvas().toBlob(function (blob) {
+      var formData = new FormData();
+      formData.append('croppedImage', blob);
+      handler(blob)
+    });
+    imgHandler(true)
+    setSaved("Saved!")
+  };
+
+  return (
+    <div>
+      <div style={{ width: "100%" }}>
+        <input type="file" accept="image/png, image/jpg, image/jpeg" onChange={onChange} required /> 
```

Comment:
For editing a pet I prefer not to have the pet picture required. can you take the value from props?